compiler/rustc_index/src/vec.rs

```diff
@@ -124,15 +124,15 @@ macro_rules! newtype_index {
 
             #[inline]
             $v const fn from_usize(value: usize) -> Self {
-                assert!(value <= ($max as usize));
+                [()][(value > ($max as usize)) as usize];
```

Comment:
Could you add ``// FIXME: replace with `assert!(value <= ($max as usize));` once `const_panic` is stable`` here and below?
Otherwise it's not obvious what's happening here.